src/ncp/ncp_uart.cpp

```diff
@@ -176,7 +176,7 @@ void NcpUart::EncodeAndSendToUart(void)
 
     if (len > 0)
     {
-        otPlatUartSend(mUartBuffer.GetBuffer(), len);
+        assert(otPlatUartSend(mUartBuffer.GetBuffer(), len) == kThreadError_None);
```

Comment:
Should we put the `assert` on a separate line? I'm not sure about every platforms implementation, but sometimes I have seen platforms that don't support it, just define it as an empty `#define assert(X) `. While not very good practice to define it that way, it would be possible (and easily overlooked) and cause this code not to even execute.